src/main/java/org/opensearch/neuralsearch/util/HybridQueryUtil.java

```diff
@@ -24,48 +26,58 @@ public class HybridQueryUtil {
-            https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L367-L370.
-            main reason for that is performance optimization, at time of writing we are ok with loosing on performance if that's unblocks
-            hybrid query for indexes with nested field types.
-            in such case we consider query a valid hybrid query. Later in the code we will extract it and execute as a main query for
-            this search request.
-            below is sample structure of such query:
+        } else if (isWrappedHybridQuery(query)) {
+            if ((hasNestedFieldOrNestedDocs(query, searchContext))) {
+                /* Checking if this is a hybrid query that is wrapped into a Bool query by core Opensearch code
+                https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/DefaultSearchContext.java#L367-L370.
+                main reason for that is performance optimization, at time of writing we are ok with loosing on performance if that's unblocks
+                hybrid query for indexes with nested field types.
+                in such case we consider query a valid hybrid query. Later in the code we will extract it and execute as a main query for
+                this search request.
+                below is sample structure of such query:
 
-            Boolean {
-               should: {
-                   hybrid: {
-                       sub_query1 {}
-                       sub_query2 {}
+                Boolean {
+                   should: {
+                       hybrid: {
+                           sub_query1 {}
+                           sub_query2 {}
+                       }
                    }
-               }
-               filter: {
-                   exists: {
-                       field: "_primary_term"
+                   filter: {
+                       exists: {
+                           field: "_primary_term"
+                       }
                    }
-               }
+                }
+                TODO Need to add logic for passing hybrid sub-queries through the same logic in core to ensure there is no latency regression */
+                // we have already checked if query in instance of Boolean in higher level else if condition
+                return ((BooleanQuery) query).clauses()
+                    .stream()
+                    .filter(clause -> clause.getQuery() instanceof HybridQuery == false)
+                    .allMatch(
+                        clause -> clause.getOccur() == BooleanClause.Occur.FILTER
+                            && clause.getQuery() instanceof FieldExistsQuery
+                            && SeqNoFieldMapper.PRIMARY_TERM_NAME.equals(((FieldExistsQuery) clause.getQuery()).getField())
+                    );
+            } else if (hasAliasFilter(query, searchContext)) {
```

Comment:
rather than if else I think we should have if cases for different conditions. I want to know the case when we have both nested fields and alias filter in the query what we will happen?

Reply:
I didn't test this scenario, let me check how the query will be constructed by core. I hope it's not double wrapped, but both filters are added to one parent bool query.

Reply:
You were right @navneet1v , we're not handling scenario with both nested fields and index alias filter properly. Basically nested fields that is first check that catches the query, kicks in and then query is failing. I'm reworking that logic now, in case of either of nested fields or alias filter or both we'll be doing same: get all filters from the parent bool and apply all those filters to every sub-query of the hybrid query. 
Example: if both nested and alias filter are there system sends us this query
```
  bool : {
     must: [
       hybrid_query
     ],
     filter: {
       alias_filter1,
       alias_filter2,
       field_exists (added for nested field)
     }
  }
```
we'll rewrite it to following form:
```
 hybrid: {
      queries: [
          bool : {
              must: [
                 <sub_query_1>
               ],
               filter: {
                  alias_filter1,
                  alias_filter2,
                  field_exists (added for nested field)
                }
          },
          bool : {
              must: [
                 <sub_query_2>
               ],
               filter: {
                  alias_filter1,
                  alias_filter2,
                  field_exists (added for nested field)
                }
          }
      ]
 }
```

Reply:
Good we are able to catch this. :) Lets fix this. Can we somehow build our code in a fashion that we don't face this issue again. :D